src/main/java/Nimbus/Nimbus.java

```diff
@@ -0,0 +1,163 @@
+package Nimbus;
+
+import Nimbus.Task.Deadline;
+import Nimbus.Task.Event;
+import Nimbus.Task.Task;
+import Nimbus.Task.Todo;
+import java.util.ArrayList;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Nimbus {
+    private final ArrayList<Task> tasks;
+    private int numTasks;
+
+    public Nimbus(ArrayList<Task> tasks, int numTasks) {
+        this.tasks = tasks;
+        this.numTasks = numTasks;
+    }
+
+    private static void greet() {
+        String logo = """
+                 _   _    _____    __  __   ____     _   _    ____ \s
+                | \\ | |  |_   _|  |  \\/  |  | __ )  | | | |  / ___|\s
+                |  \\| |    | |    | |\\/| |  |  _ \\  | | | |  \\___ \\\s
+                | . ` |   _| |_   | |  | |  | |_) | | |_| |   ___) |
+                |_| \\_|  |_____|  |_|  |_|  |____/   \\___/   |____/
+                """;
+        Nimbus.drawLine();
+        System.out.println(logo);
+        System.out.println("""
+                Hello! I'm Nimbus
+                What can I do for you?
+                """);
+        Nimbus.drawLine();
+    }
+
+    private static void exit() {
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+    private static void drawLine() {
+        System.out.println("-----------------------------------------------------");
+    }
+
+    private static void help() {
+        String help = """
+                todo [task] -- add a task without a deadline
+                deadline [task] /by [time] --add a task with a deadline
+                event [task] /from [time] /to [time] --add a task with specific start and end time
+                list --list all tasks
+                mark [x] --mark task x as done
+                unmark [x] --mark task x as not done
+                delete [x] --delete task x
+                save --save tasks to txt file
+                bye --exit""";
+        System.out.println(help);
+    }
+
+    public void addToDo(String s) {
+        this.tasks.add(this.numTasks, new Todo(s));
+    }
+
+    public void addDeadline(String s) {
+        this.tasks.add(this.numTasks, new Deadline(s));
+    }
+
+    public void addEvent(String s) {
+        this.tasks.add(this.numTasks, new Event(s));
+    }
+
+    public void addTask(String command, String arg) {
+        switch (command) {
+        case "todo" -> this.addToDo(arg);
+        case "deadline" -> this.addDeadline(arg);
+        case "event" -> this.addEvent(arg);
+        }
```

Comment:
Case statements should be of the format:
```java
// from java coding standard website
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}

```

Reply:
@fivenames hey!! just updating here, i realised these are lambda-style case statements and so this should be fine :) sorry about that